lib/purserl.ex

```diff
@@ -527,13 +507,14 @@ defmodule DevHelpers.Purserl do
               ""
           end
 
-        (Color.cyan() <>
-           error_code <> Color.reset() <> " " <> tag <> " " <> modu <> "\n") <>
-          "\n" <>
-          ((message |> add_prefix_if_missing("  ") |> syntax_highlight_indentex_lines("    ")) <>
-             "\n") <>
-          Enum.join(snippets, "\n\n") <>
-          "\n\n"
+        (case snippets do
+           # If it's more than 2 spans, we just give up because that much context is too noisy
+           [a] -> a
+           [a, b] -> a <> "\n\n" <> b
+           _ -> "<<< not printing spans, because there's more than 2. Complain to ET if this ever happens >>"
+         end <> Color.cyan() <> error_code <> Color.reset() <> " " <> tag <> " " <> modu <> "\n\n") <>
+          prefix_all_lines_if_missing("  ", message) <>
```

Comment:
```suggestion
           # If it's more than 2 spans, we just give up because that much context is too noisy
           [a] -> a
           [a, b] -> a <> "\n\n" <> b
           _ -> "<<< not printing spans, because there's more than 2. Complain to ET if this ever happens >>>"
         end <> Color.cyan() <> error_code <> Color.reset() <> " " <> tag <> " " <> modu <> "\n\n") <>
          prefix_all_lines_if_missing("  ", message) <>
```